src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationRecordPattern.java

```diff
@@ -0,0 +1,100 @@
+//non-compiled with javac: Compilable with Java21                           //indent:0 exp:0
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;     //indent:0 exp:0
+
+
+/* Config:                                                                  //indent:0 exp:0
+ * basicOffset = 4                                                          //indent:1 exp:1
+ * braceAdjustment = 0                                                      //indent:1 exp:1
+ * caseIndent = 4                                                           //indent:1 exp:1
+ * tabWidth = 4                                                             //indent:1 exp:1
+ * throwsIndent = 8                                                         //indent:1 exp:1
+ * forceStrictCondition = true                                              //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+public class InputIndentationRecordPattern {  //indent:0 exp:0
+    record ColoredPoint(boolean p, int x, int c) { } //indent:4 exp:4
+    record Rectangle(ColoredPoint upperLeft, ColoredPoint lowerRight) { } //indent:4 exp:4
+
+    void test(Object obj) { //indent:4 exp:4
+        boolean b = obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_);  //indent:16 exp:12 warn
+        boolean c = obj instanceof //indent:8 exp:8
+            ColoredPoint(_,_,_);  //indent:12 exp:12
+
+        boolean bb = obj instanceof //indent:8 exp:8
+        ColoredPoint(_,_,_);  //indent:8 exp:12 warn
+
+        if (obj instanceof ColoredPoint(_,_,_)) {} //indent:8 exp:8
+
+        if (obj instanceof //indent:8 exp:8
+                ColoredPoint(_,_,_)) {} //indent:16 exp:12 warn
+        if (obj instanceof     //indent:8 exp:8
+            ColoredPoint(_,_,_)) {} //indent:12 exp:12
+
+        if (obj instanceof  //indent:8 exp:8
+        ColoredPoint(_,_,_)) {  } //indent:8 exp:12 warn
+
+        if (obj instanceof //indent:8 exp:8
+ColoredPoint(_,_,_)) { }  //indent:0 exp:12 warn
+        if (obj instanceof //indent:8 exp:8
+ColoredPoint( //indent:0 exp:12 warn
+_,  //indent:0 exp:12 warn
+_,  //indent:0 exp:12 warn
+_)) {} //indent:0 exp:12 warn
+
+        if (obj instanceof //indent:8 exp:8
+            ColoredPoint( //indent:12 exp:12
+            _,  //indent:12 exp:12
+            _,  //indent:12 exp:12
+            _)) {} //indent:12 exp:12
+
+        if (obj instanceof ColoredPoint( //indent:8 exp:8
+            _,  //indent:12 exp:12
+            _,  //indent:12 exp:12
+            _)) {} //indent:12 exp:12
+
```

Comment:
IMO, This is a bit odd I feel like the unnamed patterns should be indented further we can treat it as a `RECORD_PATTERN_DEF` child. So I think it should be indented a bit after the pattern. like the pattern is indented a bit after ` if`

Reply:
Show a snippet of what you think better indentation is.

I don't see an issue here since this is a wrapping of the previous line and all the `_,` are on the same level.

Reply:
```
        if (obj instanceof 
            ColoredPoint( //indent:12 exp:12
                _,  //indent:16 exp:16
                _,  //indent:16 exp:16
                _)) {} //indent:16 exp:16
```
I wanted it to be like this so basically a double-line wrapping yes, because it is nested inside the record pattern not the `if`.

> But indentation check has issues with double line wrapping. It doesn't understand nested expressions should be indented a second time and only do them one time.

Then I think the current behavior is acceptable if we can't achieve a double line wrapping

Reply:
I'm ok to leave it as it is for now.